Add functionality to the Reservation screen

Axw3ll opened the pull request:
In this issue, we added functionality to the FlatList component to be able to view the flights stored in the database. We achieved this by querying Firebase and specifying that we wanted the data related to the current user's ID. We used the useEffect hook to ensure that the query is only executed once and not on every render.

# Description
-Introduces the GetFlights function which retrieves flights for the current user. It uses Firestore and Authentication from  Firebase. It imports necessary modules, defines GetFlights as an async function, fetches the current user, queries the 'flights' Firestore collection filtered by user's UID, retrieves the matching flights, and stores them in an array. It then logs and returns the flights array. Errors during execution are logged and rethrown.


## Screenshots
![imagen](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-equipo-alfa-buena-maravilla-onda/assets/88905195/50b3561b-9e75-4cc2-82ea-b7336053a779)

![imagen](https://github.com/BrightCoders-Institute/reto-reservacion-de-vuelos-equipo-alfa-buena-maravilla-onda/assets/88905195/7bde0001-02bc-4627-9d8e-d36e3e7994ff)

Axw3ll committed:
Add functionality to the Reservation screen

In this issue, we added functionality to the FlatList component to be
able to view the flights stored in the database. We achieved this by
querying Firebase and specifying that we wanted the data related to
the current user's ID. We used the useEffect hook to ensure that the
query is only executed once and not on every render.

Co-authored-by: Tole15 <rodrigermoon@gmail.com>
Co-authored-by: BrandonVG <brandon.vadillox3@gmail.com>

Eduardoespl reviewed (approved):
LGTM! 🚀

Lizzflores reviewed (approved):
LGTM!👍

xFerchoVJ commented on src/components/Card.jsx:
```diff
@@ -1,29 +1,34 @@
-const Card = () => {
+
+
+//date
+//destinationCity
+//destinationCountry
+//originCity
+//originCountry
+//passengers
```
Try not to leave comments on your PR

BrandonVG replied:
We deleted the comments

xFerchoVJ reviewed (commented):
LGTM, just remove the comments :)

BrandonVG committed:
Removed comments in files

Card.jsx:
  - Removed the comments
GetFlights.js
  - Removed a comment

Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>
Co-authored-by: Tole15 <rodrigermoon@gmail.com>

YilKanda commented on src/components/Card.jsx:
```diff
@@ -1,29 +1,27 @@
     </View>
-    
   )
 }
```
Check the indentation of your file

YilKanda commented on src/components/Card.jsx:
```diff
@@ -33,11 +31,14 @@ const styles = StyleSheet.create({
     },
     containerInfoCard:{
         backgroundColor:'#FFF',
-        width:'100%',
+        width:'90%',
         height:120,
         flexDirection:'row',
+        alignSelf:'center',
         justifyContent:'space-between',
         marginTop:8,
+        borderBottomColor:'gray',
+        borderBottomWidth:2
     },
     infOne:{
         width:'40%',
```
The styles must be in a separate file.

YilKanda commented on src/components/FlightList.jsx:
```diff
@@ -1,17 +1,36 @@
-        <FlatList />
-    )
+const FlightList = () => {
+    const [flights, setFlights] = useState([])
+    useEffect(() =>{
+        const getFlightsData = async () => {
+            const getFlights = await GetFlights();
+            setFlights(getFlights);
+        }
+        getFlightsData();
+    },[]);
```
Check the indentation

YilKanda commented on src/hooks/GetFlights.js:
```diff
@@ -0,0 +1,25 @@
+      const flight = documentSnapshot.data();
+      flights.push(flight);
+    });
+    console.log(flights);
```
Don't leave debug code on GitHub, like console.log(), you can use Alert.alert() instead

YilKanda commented on src/hooks/GetFlights.js:
```diff
@@ -0,0 +1,25 @@
+    return flights;
+  }
+  catch(error){
+    console.log(error);
```
Don't leave debug code on GitHub, like console.log(), you can use Alert.alert() instead

YilKanda commented on src/screens/HomeScreen.jsx:
```diff
@@ -8,12 +8,13 @@ import handleLogOut from '../hooks/HandleLogOut';
+  const navigation = useNavigation();  
 
   const handleOnLogOut = async () => {
     console.log('handleOnLogOut executed');
```
Don't leave debug code on GitHub, like console.log(), you can use Alert.alert() instead

Tole15 committed:
Resolve requested changes

Fixed indentation in src/components/Card.jsx.
Moved styles from src/components/Card.jsx to a separate file.
Fixed indentation in src/components/FlightList.jsx.
Removed console.log() statements from src/hooks/GetFlights.js.
Removed console.log() statement from src/screens/HomeScreen.jsx.
@YilKanda: Requested changes have been addressed. Thank you for your feedback.

Co-authored-by: BrandonVG <brandon.vadillox3@gmail.com>
Co-authored-by: Axw3ll

Tole15 committed:
Resolve requested changes

Removed console.log() statements in SignUpForm.jsx, replaced with alerts.
Improved code cleanliness and style.
Made small changes to error messages.

Co-authored-by: BrandonVG <brandon.vadillox3@gmail.com>
Co-authored-by: Axw3ll <axelomar.a.v@gmail.com>
